### ISSUE TITLE
torch 1.12 improve fairseq TransformerLayer ?

### ISSUE THREAD

**Issue by SeunghyunSEO** [facebookresearch/fairseq#4583]
According to [this link](https://pytorch.org/blog/a-better-transformer-for-fast-transformer-encoder-inference/), torch 1.12.0 improve inferece speed of [TransformerEncoder](https://pytorch.org/docs/stable/generated/torch.nn.TransformerEncoder.html), [TransformerEncoderLayer](https://pytorch.org/docs/stable/generated/torch.nn.TransformerEncoderLayer.html) or [MultiheadAttention](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html) (MHA) in specific conditions (when we use lots of padding tokens) by fusing cuda kernel so on.

However, fairseq use it's own TransformerLayer. 
Despite this, is there any improvement in fairseq too? or is it better to use pytorch's Transformer Layers?

<img width="1920" alt="2022-7-12-a-better-transformer-for-fast-transformer-encoder-inference-3" src="https://user-images.githubusercontent.com/48202736/179181455-b3111a6c-89c2-470a-aaac-d4b2cc3f4e31.png">

**Comment by gmryu** [comment]
Not now, I would believe.

Because this new feature "fast path" is applied when `why_not_fast_path==False `( `''` is False) and using [`torch._native_multi_head_attention`](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/activation.py#L1115), which is implemented with C.

Fairseq uses [`F.multi_head_attention_forward`](https://github.com/facebookresearch/fairseq/blob/main/fairseq/modules/multihead_attention.py#L538) which is the method called when `why_not_fast_path==True` (non-empty string is True).
